app/components/myissues.jsx

```diff
@@ -0,0 +1,161 @@
+var React = require("react");
+var getJSON = require("./getJSON.jsx");
+var { auth, AuthMixin } = require("./auth.jsx");
+var APIServer = "/api";
+var Labels = require("./labels.jsx");
+
+var IssueCard = React.createClass({
+  getInitialState: function() {
+    return {
+    };
+  },
+
+  getShortRepoString: function (fullUrl) {
+    var s = fullUrl;
+    s = s.replace("https://github.com/", "");
+    var i = s.indexOf("/");
+    s = s.slice(i+1);
+    i = s.indexOf("/");
+    s = s.slice(0, i);
+    return s;
+  },
+
+  render: function() {
+    var data = this.props.data;
+    var repo = this.getShortRepoString(data.html_url);
+    return (
+      <li>
+          <a href={data.html_url} target="_blank">
+              <h4>{data.title} <br/><small>{repo}</small></h4>
+              <Labels labels={data.labels} minimal='true' />
+          </a>
+      </li>
+      );
+  }
+});
+
+var ShowMyIssues = React.createClass({
+  getInitialState: function() {
+    return {
+      issuegroups: []
+    };
+  },
+  componentDidMount: function() {
+    var self = this;
+    // Assigned issues
+    getJSON(APIServer + "/myissues/assigned",
+      function(data) {
+        if (self.isMounted()) {
+          var issues = { title: 'Assigned',
+                        issues: data.issues}
+          var newState = self.state.issuegroups;
+          newState.push(issues)
+          self.setState(newState);
```

Comment:
This is interesting.

There is a bug here, but it actually happens to work anyway, took me a bit to dig it up. I'll try to explain, and I learned something while digging into this.

self.setState is being given an array, which, isn't what setState expects. Should be:

```
self.setState({
  issuegroups: newState
});
```

Now, the question is why does it work without that.

newState is a mutable array. So changing it changes the array in `self.state.issueGroups` the call to setState fails to actually update the state, but does force a rerender, which picks up the mutated issuegroups array anyway.

It would still work if you used `setState(false)` What you would get in state is `{0: false, issuesgroups: []}`

Super weird.

Looks like the react way to do this is to clone the array, make your changes, and then setState with the new array. This way bugs like the above would be obvious.